bin/generate_repos.py

```diff
@@ -191,19 +193,38 @@ def create_or_update_repo(folder, subdirectory, which_branch):
 is_team = False
 if eg is not None:
     is_team = eg.eg_team_assignment
+    # gradeable is not a vcs gradeable
+    if eg.eg_vcs_host_type == -1:
+        print((WARNING + "Warning: Semester '{}' and Course '{}' contains gradeable_id '{}' but it is not a VCS gradeable." + DEFAULT).format(args.semester, args.course, args.repo_name))
+        print("exiting")
+        sys.exit()
+# gradeable does not exist in the course
 elif not args.non_interactive:
-    print ("Warning: Semester '{}' and Course '{}' does not contain gradeable_id '{}'.".format(args.semester, args.course, args.repo_name))
-    response = input ("Should we continue and make individual repositories named '"+args.repo_name+"' for each student? (y/n) ")
-    if not response.lower() == 'y':
+    print ((WARNING + "Warning: Semester '{}' and Course '{}' does not contain gradeable_id '{}'." + DEFAULT).format(args.semester, args.course, args.repo_name))
+    # if the repo_name matches any repo_name in the different gradeables, we should ask the user if they want to continue
+    gradeables = course_connection.execute(eg_table.select()).fetchall()
+    is_repo_name_in_gradeables = False
+    for gradeable in gradeables:
+        # the eg_vcs_partial_path has pattern like `gradeable_id/user_id`, so we need to use regex to match the gradeable_id
+        if gradeable.eg_vcs_host_type == 1 and re.match(f'^{args.repo_name}/', gradeable.eg_vcs_partial_path):
+            print("Find matching gradeable_id '{}' in the course.".format(gradeable.g_id))
+            is_repo_name_in_gradeables = True
+            response = input ("Should we continue and make individual repositories named '"+args.repo_name+"' for each student/team? (y/n) ")
+            if not response.lower() == 'y':
+                print ("exiting")
+                sys.exit()
```

Comment:
If they push 'y' then it should fetch the necessary variables for this gradeable and then continue generating repos, not exit.

Reply:
Yes, it will continue generating repos. It will exit only when the reponse is not 'y'.